File: python/find_missing_number.py
def find_missing_number(A):
    n = len(A) + 1
    total_sum = n * (n - 1) // 2
    array_sum = sum(A)
    return total_sum - array_sum


def main():
    A = []
    print(find_missing_number(A))

    A = [0]
    print(find_missing_number(A))

    A = [0, 1, 2, 4, 5, 6]
    print(find_missing_number(A))


if __name__ == "__main__":
    main()

File: java/FindMissingNumber.java
public class FindMissingNumber {
    public static int findMissingNumber(int[] A) {
        int n = A.length + 1;
        int totalSum = n * (n - 1) / 2;
        int arraySum = 0;

        for (int num : A) {
            arraySum += num;
        }

        return totalSum - arraySum;
    }

    public static void main(String[] args) {
        int[] A = { 0, 1, 2, 4, 5, 6 };
        int missingNumber = findMissingNumber(A);
        System.out.println("The missing number is: " + missingNumber);
    }
}
